### ISSUE TITLE
Question about variance estimates

### ISSUE THREAD

**Issue by alec-tschantz** [nnaisense/max#4]
Apologies if this is a stupid question - but is your ensemble model predicting the variance of the state, or the variance of the state deltas?

Looking at the NLL loss function, it seems as though it is the variance of the state deltas:

`loss = (mu - targets) ** 2 / var + torch.log(var) `

where `targets` are the true state deltas, `mu` are the predicted state deltas, meaning `var` should be the variance of the state deltas.

But then, in `imagination.py`, it seems as though the model predicts the variance of the states, as opposed to the state deltas. For instance, the lines:

`next_state_means, next_state_vars = self.model.forward_all(self.states, actions)`

` next_states = self.model.sample(next_state_means[i, j], next_state_vars[i, j])`

suggest the variance is in state space, rather than delta space. 

Any clarification would be much appreciated!

**Comment by wjaskowski** [comment]
Thanks for asking. But think about it: the variance of states needs to equal to the variance of state deltas.
